### PR TITLE
Fixes to the NullTransport

### PR DESCRIPTION
**Pull request by jimduff-usds**
This PR adds some fixes to the NullTransport, mainly so we can continue with the test that causes the issue in Staging, while also allowing testing using NullTransport.

## Changes
- added a new Receiver in 'ignore' called HL7_NULL.   Changed HL7 back to sending via sftp.
- wired-in NullTransport, into SendFunction, so that its now being called.

## Checklist
- [X] Tested locally?
- [X] Ran `quickTest all`?
- [ ] Downloaded a file from http://localhost:7071/api/download
- [ ] Updated the release notes? 
- [X] Added a test?  (I added `./prime test --run hl7null`)
- [ ] Did you check for sensitive data, and remove any?
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Fixes 
- Related to #612 , but does not fix it.

## To Be Done
- Bug still not fixed.
-

### CODE REVIEW

File: prime-router/settings/organizations-local.yml [hunk 1]
```diff
@@ -405,7 +405,10 @@
           schemaName: tx/tx-covid-19
           format: HL7
         transport:
-          type: NULL
+          type: SFTP
```
Comment (jimduff-usds) [inline prime-router/settings/organizations-local.yml:408]: Changed this back, so we can continue testing the bug.

File: prime-router/settings/organizations-local.yml [hunk 2]
```diff
@@ -445,3 +448,17 @@
           type: REDOX
           apiKey: some_key
           baseUrl: http://redox:1080
+      - name: HL7_NULL
+        organizationName: ignore
```
Comment (jimduff-usds) [inline prime-router/settings/organizations-local.yml:452]: Added this new Receiver, to test using the NullTransport.    NullTransport does all the database stuff, but doesn't call SFTP or any transport.  Lets us distinguish between slow-downs caused by sftp vs slowdowns caused by lots of database interactions.

File: prime-router/src/main/kotlin/azure/SendFunction.kt [hunk 3]
```diff
@@ -79,6 +80,18 @@ class SendFunction(private val workflowEngine: WorkflowEngine = WorkflowEngine()
                                     actionHistory,
                                 )
                         }
+                        is NullTransportType -> {
+                            workflowEngine
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/azure/SendFunction.kt:84]: NullTransport is now getting called inside SendFunction.

File: prime-router/src/main/kotlin/cli/TestReportStream.kt [hunk 4]
```diff
@@ -61,6 +61,8 @@ Examples:
         ping("Is the reports endpoint alive and listening?"),
         end2end("Create Fake data, submit, wait, confirm sent via database lineage data"),
         merge("Submit multiple files, wait, confirm via db that merge occurred"),
+        dbconnections("Test weird problem wherein many 'sends' cause db connection failures"),
+        hl7null("The NULL transport does db work, but no transport.  Uses HL7 format"),
         // Note: 10,000 lines fake data generation took about 90 seconds on my laptop.  6Meg.
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/cli/TestReportStream.kt:66]: New Tests.

File: prime-router/src/main/kotlin/cli/TestReportStream.kt [hunk 5]
```diff
@@ -185,7 +189,7 @@ Examples:
         environment: ReportStreamEnv
     ) {
         echo("EndToEndTest of: ${environment.endPoint}")
-        val fakeItemCount = 20 // hack:  you need use a multiple of # of targetCounties
+        val fakeItemCount = allIgnoreReceivers.size * 5 // 5 to each receiver
         val file = FileUtilities.createFakeFile(
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/cli/TestReportStream.kt:193]: Cleanup

**Review by MauriceReeves-usds — APPROVED** [review]

**Pull request merged**